core/rio/jsonld/src/main/java/org/eclipse/rdf4j/rio/ndjsonld/NDJSONLDWriter.java

```diff
@@ -0,0 +1,113 @@
+package org.eclipse.rdf4j.rio.ndjsonld;
+
+import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.rio.*;
+import org.eclipse.rdf4j.rio.helpers.BasicWriterSettings;
+import org.eclipse.rdf4j.rio.helpers.BufferedGroupingRDFHandler;
+import org.eclipse.rdf4j.rio.helpers.JSONLDSettings;
+import org.eclipse.rdf4j.rio.jsonld.JSONLDWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+public class NDJSONLDWriter extends BufferedGroupingRDFHandler implements RDFWriter {
```

Comment:
I'm not sure that it's ok to extend BufferedGroupingRDFHandler because it is meant to be used as  a wrapper. This is  related to how things other than statements are processed, e.g. handleNamespace().

Reply:
Change the class to extend the AbstractRDFWriter and use internally the BufferedGroupingRDFHandler as a delegate.